src/helper.coffee

```diff
@@ -179,3 +184,68 @@ exports.deepEquals = deepEquals = (a, b) ->
   else
     return a is b
 
+_guid = 0
+exports.generateGUID = -> (_guid++).toString(16)
+
+# General quoted-string-fixing functionality, for use in various
+# language modes' stringFixer functions.
+
+# To fix quoting errors, we first do a lenient C-unescape, then
+# we do a string C-escaping, to add backlsashes where needed, but
+# not where we already have good ones.
+exports.fixQuotedString = (lines) ->
+  line = lines[0]
+  quotechar = if /^"|"$/.test(line) then '"' else "'"
+  if line.charAt(0) is quotechar
+    line = line.substr(1)
+  if line.charAt(line.length - 1) is quotechar
+    line = line.substr(0, line.length - 1)
+  return lines[0] = quoteAndCEscape looseCUnescape(line), quotechar
+
+exports.looseCUnescape = looseCUnescape = (str) ->
+  codes =
+    '\\b': '\b'
+    '\\t': '\t'
+    '\\n': '\n'
+    '\\f': '\f'
+    '\\"': '"'
+    "\\'": "'"
+    "\\\\": "\\"
+    "\\0": "\0"
+  str.replace /\\[btnf'"\\0]|\\x[0-9a-fA-F]{2}|\\u[0-9a-fA-F]{4}/g, (m) ->
+    if m.length is 2 then return codes[m]
+    return String.fromCharCode(parseInt(m.substr(1), 16))
+
+exports.quoteAndCEscape = quoteAndCEscape = (str, quotechar) ->
+  result = JSON.stringify(str)
+  if quotechar is "'"
+    return quotechar +
+      result.substr(1, result.length - 2).
+             replace(/((?:^|[^\\])(?:\\\\)*)\\"/g, '$1"').
+      replace(/'/g, "\\'") + quotechar
+  return result
+
+# A naive dictionary mapping arbitrary objects to arbitrary objects, for use in
+# ace-to-droplet session matching.
+#
+# May replace with something more sophisticated if performance becomes an issue,
+# but we don't envision any use cases where sessions flip really fast, so this is unexpected.
+exports.PairDict = class PairDict
```

Comment:
It is used in the Sessions code. I think that code must have already been merged, since it's not showing up in the diff.